google-cloud-clients/google-cloud-storage/src/test/java/com/google/cloud/storage/it/ITStorageTest.java

```diff
@@ -2442,4 +2443,69 @@ public void testGetServiceAccount() {
     assertNotNull(serviceAccount);
     assertTrue(serviceAccount.getEmail().endsWith(SERVICE_ACCOUNT_EMAIL_SUFFIX));
   }
+
+  @Test
+  public void testBucketWithBucketPolicyOnlyEnabled() throws Exception{
+    String bpoBucket = RemoteStorageHelper.generateBucketName();
+    try {
+      storage.create(
+              Bucket.newBuilder(bpoBucket)
+                      .setIamConfiguration(
+                              BucketInfo.IamConfiguration.newBuilder().setBucketPolicyOnly(
+                                      BucketInfo.BucketPolicyOnly.newBuilder()
+                                              .setEnabled(true)
+                                              .build())
+                                      .build())
+                      .build());
+
+
+      Bucket remoteBucket = storage.get(bpoBucket, Storage.BucketGetOption.fields(BucketField.IAMCONFIGURATION));
+
+      assertTrue(remoteBucket.getIamConfiguration().getBucketPolicyOnly().getEnabled());
+      assertNotNull(remoteBucket.getIamConfiguration().getBucketPolicyOnly().getLockedTime());
+      try {
+        remoteBucket.listAcls();
+      } catch (StorageException e) {
+        //Expected: Listing legacy ACLs should fail on a BPO enabled bucket
+      }
+      try {
+        remoteBucket.listDefaultAcls();
+      } catch (StorageException e) {
+        //Expected: Listing legacy ACLs should fail on a BPO enabled bucket
+      }
+
+    } finally {
+      RemoteStorageHelper.forceDelete(storage, bpoBucket, 1, TimeUnit.MINUTES);
+    }
+  }
+
+  @Test
+  public void testEnableBucketPolicyOnlyOnExistingBucket() throws Exception{
+    String bpoBucket = RemoteStorageHelper.generateBucketName();
+    try {
+
+      BucketInfo.BucketPolicyOnly bucketPolicyOnly = BucketInfo.BucketPolicyOnly.newBuilder()
+              .setEnabled(false)
+              .build();
+      BucketInfo.IamConfiguration iamConfiguration = BucketInfo.IamConfiguration.newBuilder().setBucketPolicyOnly(
+              bucketPolicyOnly).build();
```

Comment:
@JesseLovelace wdyt about moving the fields for BucketPolicyOnly under IamConfiguration instead of its class? 

At the moment the code to set the value `true`/`false` is built with three builders. It would make the following a bit shorter:

```java
Storage storage = StorageOptions.getDefaultInstance().getService();

// The name of a bucket, e.g. "my-bucket"
// String bucketName = "my-bucket";

BucketInfo.BucketPolicyOnly bucketPolicyOnly = BucketInfo.BucketPolicyOnly.newBuilder().setEnabled(true).build();
BucketInfo.IamConfiguration iamConfiguration = BucketInfo.IamConfiguration.newBuilder().setBucketPolicyOnly(
    bucketPolicyOnly).build();
Bucket bucket =
    storage.update(BucketInfo.newBuilder(bucketName).setIamConfiguration(iamConfiguration).build());

System.out.println("Bucket Policy Only was enabled for " + bucketName);
```

Into:

```java
Storage storage = StorageOptions.getDefaultInstance().getService();

// The name of a bucket, e.g. "my-bucket"
// String bucketName = "my-bucket";

BucketInfo.IamConfiguration iamConfiguration = BucketInfo.IamConfiguration.newBuilder().setBucketPolicyOnlyEnabled(true).build();
Bucket bucket =
    storage.update(BucketInfo.newBuilder(bucketName).setIamConfiguration(iamConfiguration).build());

System.out.println("Bucket Policy Only was enabled for " + bucketName);
```